src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -289,7 +325,7 @@ fn rustc(cx: &mut Context, unit: &Unit) -> CargoResult<Work> {
         }
 
         state.running(&rustc);
-        if json_messages {
+        let cont = if json_messages {
             rustc.exec_with_streaming(
```

Comment:
Yeah I'm slightly worried about this in terms of only bits and pieces of Cargo's API will work with these custom arguments. @nrc is this just avoided by the rls?